Libraries/LibWasm/AbstractMachine/BytecodeInterpreter.cpp

```diff
@@ -391,1286 +2515,361 @@ double BytecodeInterpreter::read_value<double>(ReadonlyBytes data)
     return bit_cast<double>(read_value<u64>(data));
 }
 
-ALWAYS_INLINE void BytecodeInterpreter::interpret_instruction(Configuration& configuration, InstructionPointer& ip, Instruction const& instruction)
+CompiledInstructions try_compile_instructions(Expression const& expression, Span<FunctionType const>)
 {
-    dbgln_if(WASM_TRACE_DEBUG, "Executing instruction {} at ip {}", instruction_name(instruction.opcode()), ip.value());
-
-    switch (instruction.opcode().value()) {
-    case Instructions::unreachable.value():
-        m_trap = Trap::from_string("Unreachable");
-        return;
-    case Instructions::nop.value():
-        return;
-    case Instructions::local_get.value():
-        configuration.value_stack().append(Value(configuration.frame().locals()[instruction.arguments().get<LocalIndex>().value()]));
-        return;
-    case Instructions::local_set.value(): {
-        auto value = configuration.value_stack().take_last();
-        configuration.frame().locals()[instruction.arguments().get<LocalIndex>().value()] = value;
-        return;
+    CompiledInstructions result;
+    result.dispatches.ensure_capacity(expression.instructions().size());
+
+    for (auto& instruction : expression.instructions())
+        result.dispatches.unchecked_append({ &instruction, { .sources = { Dispatch::Stack, Dispatch::Stack, Dispatch::Stack }, .destination = Dispatch::Stack } });
```

Comment:
Especially with the next commit,
couldn't the src/dst state here be just the default state of the values?
Or a at least be a named state, so to decrease the quite long lines this produces

Reply:
So then a (locally) named state to avoid repetition?